src/NuGet.Services.AzureSearch/SearchService/AuxiliaryDataCache.cs

```diff
@@ -27,20 +27,20 @@ public class AuxiliaryDataCache : IAuxiliaryDataCache
 
         public bool Initialized => _data != null;
 
-        public async Task InitializeAsync()
+        public async Task EnsureInitializedAsync()
         {
             if (!Initialized)
             {
-                await LoadAsync(Timeout.InfiniteTimeSpan, CancellationToken.None);
+                await LoadAsync(Timeout.InfiniteTimeSpan, allowNoOp: true, token: CancellationToken.None);
             }
         }
 
         public async Task TryLoadAsync(CancellationToken token)
         {
-            await LoadAsync(TimeSpan.Zero, token);
+            await LoadAsync(TimeSpan.Zero, allowNoOp: false, token: token);
         }
 
-        private async Task LoadAsync(TimeSpan timeout, CancellationToken token)
+        private async Task LoadAsync(TimeSpan timeout, bool allowNoOp, CancellationToken token)
```

Comment:
Maybe `shouldReload` or something similar?
`allowNoOp` isn't super descriptive as to what it's allowing to no-op.

Reply:
Yeah, that's better